Assets/Scripts/Weapons/Weapon.cs

```diff
@@ -13,17 +13,16 @@ public class Weapon : MonoBehaviour
         { "repeater", false }
     };
 
-    AbstractAttackBehaviour _attackBehaviour;
-    float _fireRate; // shots per second
-    int _magazineSize;
+    public AbstractAttackBehaviour _attackBehaviour;
+    public float _fireRate; // shots per second
+    public int _magazineSize;
```

Comment:
let me know if we shouldn't expose these fields, I changed them to public to get access to apply the powerups